src/providers/WorkflowCore.Persistence.RavenDB/Services/RavendbPersistenceProvider.cs

```diff
@@ -41,6 +41,11 @@ static void CreateIndexes(RavendbPersistenceProvider instance)
 
 		public async Task<string> CreateNewWorkflow(WorkflowInstance workflow, CancellationToken cancellationToken = default)
 		{
+			if (workflow.CorrelationId != null)
```

Comment:
Wouldn't it break existing solutions based on RavenDB?

Reply:
No, since any existing solution does not pass the CorrelationId parameter (i.e. CorrelationId would be null), so nothing would change and behavior would be the same as before.

If someone would try to get benefits of the CorrelationId with RavenDB persistence only then they would see an error encouraging to contribute with tests first and then similar change - add unique constraint in the database.